### PR TITLE
Remove duplicate classes topic from C# guide

### LINKED ISSUE #964: C# Guide: Classes
This issue tracks the topic for http://github.com/dotnet/core-docs/csharp/classes.md

Much of the content can be taken from here: https://msdn.microsoft.com/en-us/library/ms173109.aspx

### PR DESCRIPTION
**Pull request by pkulikov**
Combined [C# Concepts classes](https://docs.microsoft.com/en-us/dotnet/csharp/classes) article into the [Classes (C# Programming guide)](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/classes) article.

Additionally:
- Inlined small code pieces as it makes a source Markdown file more readable.
- Minor editions.

Fixes #964 

In tandem with PR dotnet/samples#8

### CODE REVIEW

**Comment by pkulikov** [comment]
@BillWagner I put WIP as I also want to move the example snippet to the samples repo.

**Comment by mairaw** [comment]
@pkulikov so we have a current work item (#1255) that we haven't been able to finish yet (it was supposed to be automated) but I manually change some PRs when I see they're touching snippets in these codesnippet folders.

The first time we migrated these topics, they erroneously broken down each sample into these tiny little snippets. But we do have the snippets in the samples repo already and you can see how the topic should have looked like if it got migrated correctly the first time here:
https://raw.githubusercontent.com/dotnet/docs/migration_csharpvb_reexport/docs/csharp/programming-guide/classes-and-structs/classes.md

So the original snippet for this probably lives here:
https://github.com/dotnet/samples/blob/master/snippets/csharp/VS_Snippets_VBCSharp/csProgGuideObjects/CS/Objects.cs

But you can probably change the samples location to ~/samples now that they don't live in this repo anymore. Something to test.

**Comment by pkulikov** [comment]
@mairaw interesting that that file with a lot of snippets doesn't contain snippets for the topic in interest; as if those were cut out of the file.

I've made the PR in the samples repo to move the snippet to programming guide snippets here:
https://github.com/dotnet/samples/tree/master/snippets/csharp/programming-guide/classes-and-structs

I assume the old way to link snippets from the Markdown files continues to work even when snippets are in the different repo.

**Comment by mairaw** [comment]
It's something we also need to understand better ourselves. It's been in my to-do list to talk to the engineering team about this workflow. When I catch a break, I'll take a look at your changes here and there. Exciting times! 😄

**Comment by pkulikov** [comment]
We need a notion of *tandem PRs* :)

**Comment by pkulikov** [comment]
@BillWagner and we should merge this PR: all other PRs do not build because of this one.

File: docs/csharp/programming-guide/classes-and-structs/classes.md [hunk 1]
```diff
@@ -14,40 +15,69 @@ author: "BillWagner"
...
-  
- Unlike structs, classes support *inheritance*, a fundamental characteristic of object-oriented programming. For more information, see [Inheritance](../../../csharp/programming-guide/classes-and-structs/inheritance.md).  
+A *class* is a construct that enables you to create your own custom types by grouping together variables of other types, methods and events. A class is like a blueprint. It defines the data and behavior of a type. If the class is not declared as static, client code can create *instances* of a class or *objects* which are assigned to a variable. The instance of a class remains in memory until all references to it go out of scope. At that time, the CLR marks it as eligible for garbage collection. If the class is declared as [static](../../../csharp/language-reference/keywords/static.md), then only one copy exists in memory and client code can only access it through the class itself, not an *instance variable*. For more information, see [Static Classes and Static Class Members](../../../csharp/programming-guide/classes-and-structs/static-classes-and-static-class-members.md).
```
Comment (BillWagner) [inline docs/csharp/programming-guide/classes-and-structs/classes.md]: There are a few sentences here that could be more clear:

> If the class is not declared as static, client code can create *instances* of a class or *objects* which are assigned to a variable.  

Consider: "If the class is not declared as static, client code can create *instances* of it. These instances are *objects* which are assigned to a variable."

> If the class is declared as [static](../../../csharp/language-reference/keywords/static.md), then only one copy exists in memory and client code can only access it through the class itself, not an *instance variable*. 

Consider: "If the class is declared as [static](../../../csharp/language-reference/keywords/static.md), you cannot create instances, and client code can only access it through the class itself."

**Review by BillWagner — CHANGES_REQUESTED** [review]
@pkulikov Overall, this is a great improvement to combine these two topics. There were a couple sentences that could be misinterpreted. I offered suggestions for other wording.  Once that's done, I'll :shipit:

Thanks again.

**Comment by pkulikov** [comment]
@BillWagner thank you for feedback, I've updated wording as you've suggested.

**Review by BillWagner — APPROVED** [review]

**Comment by mairaw** [comment]
If you approved this @BillWagner, I'll merge it then because all builds are breaking now because the samples change is in but not this one.

**Pull request merged**